src/sample/Program.cs

```diff
@@ -1,96 +1,94 @@
-﻿using Azure.Core.Diagnostics;
-using Azure.Identity;
+﻿using Azure.Identity;
 using DevLab.JmesPath;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
 using Microsoft.Graph.Cli.Core.Authentication;
 using Microsoft.Graph.Cli.Core.Commands.Authentication;
 using Microsoft.Graph.Cli.Core.Configuration;
 using Microsoft.Graph.Cli.Core.Configuration.Extensions;
 using Microsoft.Graph.Cli.Core.Http;
 using Microsoft.Graph.Cli.Core.IO;
+using Microsoft.Kiota.Abstractions;
 using Microsoft.Kiota.Abstractions.Authentication;
 using Microsoft.Kiota.Authentication.Azure;
 using Microsoft.Kiota.Cli.Commons.IO;
 using Microsoft.Kiota.Http.HttpClientLibrary;
 using System;
 using System.Collections.Generic;
 using System.CommandLine;
+using System.CommandLine.Binding;
 using System.CommandLine.Builder;
 using System.CommandLine.Hosting;
+using System.CommandLine.Invocation;
 using System.CommandLine.IO;
 using System.CommandLine.Parsing;
-using System.Diagnostics.Tracing;
 using System.IO;
-using System.Linq;
 using System.Reflection;
-using System.Security.Cryptography;
 using System.Threading.Tasks;
 
 namespace Microsoft.Graph.Cli
 {
+    class ServiceBinder<T> : BinderBase<T?> where T: class
+    {
+        protected override T? GetBoundValue(BindingContext ctx)
+        {
+            var host = ctx.GetRequiredService(typeof(IHost)) as IHost;
+
+            return (ctx.GetService(typeof(T)) ?? host?.Services.GetService(typeof(T))) as T;
```

Comment:
can you expand on why we need to look for the value at two different places?

Reply:
This is a sample showing how one can use both the .NET Dependency injection library and the System.CommandLine container depending on where they registered the type. Notice that T is a generic type here.